examples/crate.elm

```diff
@@ -143,42 +170,120 @@ view { texture, theta } =
             []
 
         Just tex ->
-            [ render vertexShader fragmentShader crate { crate = tex, perspective = perspective theta } ]
+            let
+                camera =
+                    perspective theta
+            in
+                [ renderBox
+                    [ disable stencilTest ]
+                    Math.Matrix4.identity
+                    (vec3 1 1 1)
+                    tex
+                    camera
+                , renderFloor
+                    [ enable stencilTest
+                    , stencilFunc WebGL.Constants.always 1 255
+                    , stencilOperation keep keep replace
+                    , stencilMask 1
+                    , depthMask 0
```

Comment:
Both `depthMask` and `ColorMask` currently take an `Int`, when they should take a `Bool`. These are just for toggling stuff on or off.
https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/colorMask / https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/depthMask.

`stencilMask` is a real mask and has to take an int: https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/stencilMask